src/autorun.ts

```diff
@@ -27,11 +34,28 @@ export function run<T>(fn: Cb<T>): Observable<T> {
     const $ = createTracker(true);
     const _ = createTracker(false);
 
+    let error: any;
+    let hasError = false;
+
+    const anyDepRunning = () => {
+        for (let dep of deps.values()) {
+            if (dep.track && !dep.completed) {
+                // One of the $-tracked deps is still running
+                return true;
+            }
+        }
+        // All $-tracked deps completed
+        return false;
+    }
+
     return new Observable((observer) => {
         const sub = update$
             .pipe(
                 startWith(void 0), // run fn() instantly
                 switchMap(() => {
+                    if (hasError) {
+                        return throwError(error);
+                    }
```

Comment:
**a thought to discuss, do not implement it in this PR**

an idea: maybe in future we could `throw` right inside the expression in the read-value phase
```ts
const o = of(1); // some observable
const e = throwErr(''); // erroneous observable
run(() => $(o) + $(e));
//               ^ throw here
// so that we could handle it in-place
run(() => {
  try{
    return $(o) + $(e);
  } catch(err) {
    return 'handled error';
  }
);
```
what do you think?

**again, this comment is just for the sake of discussion 🙂**

Reply:
For the use cases I'm thinking of, I'm not so sure how this can be useful... I'd like it to have more or less the semantics of other operators where the selector functions don't have to deal with errors, except for the operators that are built specifically for this purpose.
Also what would the semantics be when you handled an error? Will it resubscribe to the failing observable to try again? Will it emit 'handled error' and then complete?
I think it's better to leave errors up to the operators that are built for it (like `catchError()`, `retryWhen()` etc), to be able to specifically say how to continue after an error, but that's just my 2ct :)

Reply:
I don't have a use-case for this. It just *looks* right, idk. If we have synchronous-like code in expressions `$(a) + $(b)`, then the error should be also thrown in synchronous fashion. And since unhandled errors would error to the stream -- in practice it would equal your current implementation.

While I agree that errors should be handled operator-wise and it's really not clear what to do after that with the streams... 🙂

Thanks for your feedback!
I guess, I'll put this idea to sleep for the time being 😴 🤷‍♂️